src/Illuminate/Auth/Access/Admission.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Illuminate\Auth\Access;
+
+class Admission
+{
+    /**
+     * Indicates whether admission was allowed.
+     *
+     * @var bool
+     */
+    protected $allowed;
+
+    /**
+     * The reason admission was allowed/denied.
+     *
+     * @var string|null
+     */
+    protected $reason;
+
+    /**
+     * Create a new admission instance.
+     *
+     * @return void
+     */
+    protected function __construct()
+    {
+        //
+    }
+
+    /**
+     * Create an allowed admission.
+     *
+     * @param  string|null  $reason
+     * @return static
+     */
+    public static function allow($reason = null)
+    {
+        $admission = new static;
+
+        $admission->allowed = true;
+        $admission->reason = $reason;
+
+        return $admission;
+    }
+
+    /**
+     * Create a denied admission.
+     *
+     * @param  string  $reason
+     * @return static
+     */
+    public static function deny($reason = 'This action is unauthorized.')
```

Comment:
why not use `null` as a default - php and symfony both do this as a convention for defaults, making it easy to skip args